tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/decisiontree/DecisionTree.scala

```diff
@@ -17,8 +17,8 @@ import scala.collection.mutable
   * appearance at that node (i.e. how many times a training vector with
   * that outcome makes it to this node during classification)
   */
-case class DecisionTree(outcomes: Iterable[Int], child: IndexedSeq[Seq[(Int, Int)]],
-  splittingFeature: IndexedSeq[Option[Int]], outcomeHistograms: IndexedSeq[Seq[(Int, Int)]])
+case class DecisionTree(outcomes: Iterable[Int], child: IndexedSeq[Map[Int, Int]],
```

Comment:
Are you constructing these maps as [IntMap](http://www.scala-lang.org/api/current/index.html#scala.collection.immutable.IntMap)? You should try - it might substantially reduce your memory overhead.